src/App.tsx

```diff
@@ -1,27 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadGoods = async (params:() => Promise<Good[]>) => {
+    setIsLoading(true);
+    try {
+      const goods = await params();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      setVisibleGoods(goods);
+    } catch {
+      setIsLoading(false);
+    }
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => loadGoods(getAll)}
+      >
+        Load all goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => loadGoods(get5First)}
+      >
+        Load 5 first goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="red-button"
+        onClick={() => loadGoods(getRedGoods)}
+      >
+        Load red goods
+      </button>
+
+      {isLoading && <GoodsList goods={visibleGoods} />}
```

Comment:
It looks like loading goods in progress, and you render goods lists -> sounds weird, doesn't it?
It's rather some checking on a error
```suggestion
      {isLoading && <GoodsList goods={visibleGoods} />}
```

Reply:
Let's rename `isLoading` variable